spec/client_spec.rb

```diff
@@ -115,5 +115,125 @@
         client.address.retrieve('adr_123')
       }.to raise_error(EasyPost::Errors::NotFoundError) # should throw error because our lambda returns 404
     end
+
+    describe 'hooks' do
+      after { EasyPost::Hooks.send(:subscribers).clear }
+
+      it 'subscribes to request events' do
+        notifications = []
+        client = described_class.new(api_key: ENV['EASYPOST_TEST_API_KEY'])
+        client.subscribe_request_hook do |request_data|
+          expect(request_data).to be_an(EasyPost::Hooks::RequestContext)
+          expect(request_data.method).to eq(:get)
+          expect(request_data.path).to end_with('/addresses/adr_123')
+          expect(request_data.headers).to be_a(Hash)
+          request_data.headers.each { |key, value| expect(request_data.headers[key]).to eq(value) }
+          expect(request_data.request_body).to be_nil
```

Comment:
issue: expecting the body to be nil doesn't unfortunately test that we populated it with the right values. Can we follow the other lib test suites for hooks as they have assertions that can test all of this?

Reply:
I'm testing a `GET` here, so the request body is expected to be `nil` in this case. Can add a comment clarifying that and add the assertions on the specs below which test for `POST`